sdc_client/balancer.py

```diff
@@ -12,12 +13,61 @@ class StreamsetsBalancer:
     def __init__(self):
         self.streamsets_pipelines: Dict[IStreamSets, List[IPipeline]] = get_streamsets_pipelines()
 
+    def _refresh_object(self):
+        self._preferred_types = set([ss.get_preferred_type() for ss in self.streamsets_pipelines if ss.get_preferred_type()])
+        self._pipelines = list(itertools.chain.from_iterable(self.streamsets_pipelines.values()))
+        self._streamsets = list(self.streamsets_pipelines.keys())
+        self.balanced_streamsets_pipelines: Dict[IStreamSets, List[IPipeline]] = {ss: [] for ss in self._streamsets}
+        self.rebalance_map = {}
+        self.structure = {}
+        for type_ in self._preferred_types:
+            self.structure[type_] = {}
+            self.structure[type_]['streamsets'] = [ss for ss in self._streamsets if ss.get_preferred_type() == type_]
+            self.structure[type_]['pipelines'] = [pipeline for pipeline in self._pipelines if pipeline.source_type == type_]
+        self.structure[None] = {}
+        self.structure[None]['streamsets'] = [ss for ss in self._streamsets if not ss.get_preferred_type() in self._preferred_types]
```

Comment:
Can we use more convenient here? `if ss.get_preferred_type() not in self._preferred_types` instead of `if not ss.get_preferred_type() in self._preferred_types`